File: meal-planner-api/aspnet/MealPlanner.ObjectModels/Models/NutrientModel.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MealPlanner.ObjectModels.Models
{
  /// <summary>
  /// Represents the _Nutrient_ model (a nutrient on nutrition facts label)
  /// </summary>
  public class NutrientModel : IValidatableObject//, IModel
  {
    public int NutrientTypeId { get; set; }
    
    public int? DailyValuePercentage { get; set; }

    public AmountModel Amount { get; set; }


    /// <summary>
    /// Represents the _Nutrient_ `Validate` method
    /// </summary>
    /// <param name="validationContext"></param>
    /// <returns></returns>
    public IEnumerable<ValidationResult> Validate(ValidationContext validationContext) => new List<ValidationResult>();
  }
}

File: meal-planner-client/angular/src/app/data/nutrient.model.ts
import { Amount } from './amount.model';

/**
 * Represents the _Nutrient_ model
 *
 * ```yaml
 * nutrientTypeId: number;
 * dailyValuePercentage?: number;
 * amount: Amount;
 * ```
 */
export interface Nutrient {
  nutrientTypeId: number;
  dailyValuePercentage?: number;
  amount: Amount;
}